include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm.hpp

```diff
@@ -164,63 +163,54 @@ class pairwise_alignment_algorithm
     template <std::ranges::forward_range sequence1_t, std::ranges::forward_range sequence2_t>
     int32_t compute_matrix(sequence1_t & sequence1, sequence2_t & sequence2)
     {
-        // Use gap costs locally.
-        int32_t gap_extension{m_gap_scheme.get_gap_score()};
-        int32_t gap_open{m_gap_scheme.get_gap_open_score() + gap_extension};
-
-        std::vector<int32_t> optimal_column{};
-        std::vector<int32_t> horizontal_column{};
-
-        // Initialise matrix.
-        optimal_column.resize(sequence2.size() + 1, 0);
-        horizontal_column.resize(sequence2.size() + 1, 0);
-        int32_t diagonal{};
-        int32_t vertical{gap_open};
-
-        // Initialise the first column ignoring the very first cell which was already initialised with 0.
-        for (auto && [opt, hor] : views::zip(optimal_column, horizontal_column) | views::drop(1))
+        // ---------------------------------------------------------------------
+        // Initialisation phase: allocate memory and initialise first column.
+        // ---------------------------------------------------------------------
+
+        thread_local score_matrix_single_column<int32_t> local_matrix{};
+        local_matrix.reset_matrix(std::forward<sequence1_t>(sequence1), std::forward<sequence2_t>(sequence2));
+        auto matrix_iterator = local_matrix.begin();
+        auto alignment_column = *matrix_iterator;
+
+        // Initialise the first column.
+        *alignment_column.begin() = this->initialise_origin_cell();
+        for (auto cell : alignment_column | seqan3::views::drop(1))
+            cell = this->initialise_first_column_cell(cell);
+
+        // ---------------------------------------------------------------------
+        // Recursion phase: compute column-wise the alignment matrix.
+        // ---------------------------------------------------------------------
+
+        // Compute remaining matrix
+        ++matrix_iterator; // Move to next column.
+        for (auto it_col = sequence1.begin(); it_col != sequence1.end(); ++it_col, ++matrix_iterator)
```

Comment:
your iterator naming is a little confusing. There is the `_iterator` suffix, the `_iter` suffix and the `it_` prefix in this for loop.
Do you mind making them consistent one way or the other?

Reply:
yes, I hate iterator naming.